lib/avro/builder/dsl.rb

```diff
@@ -89,6 +89,25 @@ def type(*)
         @last_object = super
       end
 
+      # Several variants:
+      # define_type(:foo, long)
+      # define_type(:bar, { namespace: 'com.example' }, long)
```

Comment:
Actually, I'm going to reverse what I just said. I dislike the name and namespace arguments being separated, but having namespace come as part of a final options argument is consistent with other parts of the DSL. For other named types, the name comes first and then the namespace is part of the options. For a field that uses a named type the ordering is name, type, then options which may include `:type_namespace`.